### PR TITLE
feat: adding plan to staging and production runs

### PR DESCRIPTION
**Pull request by QuanNguyen-CDS**
# Summary | Résumé
-added a new version file called `infrastructure_version.txt` we need this because we are sharing the configuration between `merge_to_main_production.yml` and `pull_requests.yml` and only want one place to edit
- moved the serialization of the INFRASTRUCTURE_VERSION into a step in `merge_to_main_production.yml`
- added testing of production files to determine if we are running a plan for production or staging
- remove selective module `terragrunt plan` based upon the files/modules touched within the PR

# Test instructions | Instructions pour tester la modification
- Tested that pull_request now does production plan, it automatically runs production now because of the file changes within this PR
- Tested that pull_request now does staging plans, manually modified the pull_request.yml to force staging
- Looked within the pull request run to see that INFRASTRUCTURE_VERSION, WORKFLOWS, are properly filled in

# Notes | Remarques
- Have NOT tested INFRASTRUCTURE_VERSION serialization on `merge_to_main_production.yml`, the INFRASTRUCTURE_VERSION however has been testing during the plan testing within `pull_requests.yml` and it's the same code copy and pasted into `merge_to_main_production.yml`
- Since this is a special PR, the pull_request run will be on production because the PR introduces the new infrastructure_version.txt file, which indicates to the pull_request GH actions that it's touching production
- Did NOT test for IS_PRODUCTION as no changes were done to env/production/*

# Reviewer checklist | Liste de vérification du réviseur
- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: .github/workflows/infrastructure_version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+0.87.0
```
Comment (AntoineAugusti) [inline .github/workflows/infrastructure_version.txt:1]: Can you update the README to reference this file?

File: .github/workflows/pull_requests.yml [hunk 2]
```diff
@@ -17,16 +17,51 @@ env:
...
 
+      - name: Get changed files
+        id: changed-files
+        uses: jitterbit/get-changed-files@v1
+        with:
+          format: 'json'
+
+      - name: Get touched GitHub Actions configurations
+        run: |
+          WORKFLOWS=`echo '${{ steps.changed-files.outputs.added_modified }}' | jq -c '[.[] | match("(.*github?)\/(.*workflows?)\/(.*)").captures[2].string] | unique | select(length > 0)'`
```
Comment (AntoineAugusti) [inline .github/workflows/pull_requests.yml:40]: Is this regex right? You're capturing the 2nd group but you're not including file extension after. My understanding is that you'll get `merge_to_main_production.yml` and not `merge_to_main_production`. Does `contains` work with that?

`.` is not escaped

File: .github/workflows/pull_requests.yml [hunk 3]
```diff
@@ -17,16 +17,51 @@ env:
...
+          echo "WORKFLOWS=$WORKFLOWS" >> $GITHUB_ENV
+
+      - name: Get touched production Terragrunt configurations
+        run: |
+          IS_PRODUCTION=`echo '${{ steps.changed-files.outputs.added_modified }}' | jq -c '[.[] | match("(.*env?)\/(.*)\/(.*)\/").captures[1].string] | unique | select(length > 0)'`
```
Comment (AntoineAugusti) [inline .github/workflows/pull_requests.yml:45]: Is this regex right? The first group seems to match anything ending with `env` whereas paths should be `env/{production|staging}`. Could you change to a simpler regex?

File: .github/workflows/pull_requests.yml (hunk 2 above)
Comment (QuanNguyen-CDS) [reply .github/workflows/pull_requests.yml:40]: addressed by adding file extensions

**Review by AntoineAugusti — APPROVED** [review]
LGTM, thanks for working on this!

**Pull request merged**